infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -301,21 +313,85 @@ public void setSnapshotApplied(LogReplicationEntry entry) {
                 " persistSiteConfigID " + persistSiteConfigID + " persistSnapStart " + persistSnapStart);
     }
 
+    public void setReplicationRemainingPercent(long percentComplete) {
+        ReplicationStatusKey key = ReplicationStatusKey.newBuilder().setKey(
+                ReplicationStatusKey.Status.REPLICATION_COMPLETION).build();
+        ReplicationStatusVal val = ReplicationStatusVal.newBuilder().setReplicationCompletion(percentComplete).build();
+        TxBuilder txBuilder = corfuStore.tx(NAMESPACE);
+        txBuilder.update(REPLICATION_STATUS_TABLE, key, val, null);
+        txBuilder.commit();
+    }
+
+    public long getReplicationRemainingPercent() {
+        ReplicationStatusKey key = ReplicationStatusKey.newBuilder().setKey(
+                ReplicationStatusKey.Status.REPLICATION_COMPLETION).build();
+        CorfuRecord record = corfuStore.query(NAMESPACE).getRecord(REPLICATION_STATUS_TABLE, key);
+        if (record == null) {
+            return 0;
```

Comment:
maybe we should add a log here so we know that nothing has been updated to the corfuStore and not that it is complete necessarily.